src/storage.ts

```diff
@@ -44,13 +48,13 @@ const storageProxyHandler: ProxyHandler<PersistantStorage> = {
       }
     }
     if (target.cache[propKey]) return target.cache[propKey];
-    try {
-      const data = target.retrieve(propKey);
+
+    return target.retrieve(propKey).then((data) => {
```

Comment:
can we change this to use async await format?

Reply:
when u change this getter to async function u wont be able to call retrieve,clear and other functions not only that but sometimes the getter return the cached instance instead of getting from storage that's why async function wasn't  used here